Pseudocode/Stack/Stock Span/SourceCode.tex

```diff
@@ -0,0 +1,67 @@
+% Set the Page Layout
+\documentclass[12pt]{article}
+\usepackage[inner = 2.0cm, outer = 2.0cm, top = 2.0cm, bottom = 2.0cm]{geometry}
+
+
+% Package to write pseudo-codes
+\usepackage{algorithm}
+
+% Don not remove the 'end' at the end of the algorithm
+\usepackage{algpseudocode}
+\usepackage{xcolor}
+
+% Let us keep the 'end only for 'For'
+% Manually remove the 'end' for some sections
+\algtext*{EndIf}
+\algtext*{EndFunction}
+
+% Define Left Justified Comments
+\algnewcommand{\LeftComment}[1]{\Statex \(\triangleright\) #1}
+
+% Remove the Numbering of the Algorithm
+\usepackage{caption}
+\DeclareCaptionLabelFormat{algnonumber}{Algorithm}
+\captionsetup[algorithm]{labelformat = algnonumber}
+
+\newcommand{\IsNotEmpty}{is \textbf{not empty, }}
+\newcommand{\Print}{\textbf{Print}}
+\newcommand{\To}{\textbf{ to }}
+\newcommand{\And}{\textcolor{violet}{and}}
+
+\begin{document}
+
+    \begin{algorithm}
+    
+      \caption{Stock Span}
+      
+      \begin{algorithmic}[1]
+        \Ensure One Based Indexing
+        \Statex
+        \Function {$Calculate\_Stock\_Span$}{$price$}
+            \LeftComment{$span[i]$ denotes the span of the price of stock at the $i-th$ day}
+            
+            \Statex
+            \State $len \gets price.length$
+            \For{$ i = 1 \To len$}
+                \While{$stack$ \IsNotEmpty \ \And \ $price[stack.top] \leq price[i]$}
+                    \State $stack.pop$
+                \EndWhile
+                
+                \Statex
+                \If{$stack$ \IsNotEmpty}
+                    \State $span[i] \gets (i - stack.top)$
+                \Else 
+                    \State $span[i] \gets i+1$
```

Comment:
Serious bug here. If you are using one based indexing, then it should be `i` and not `i+1`

Reply:
Also include a right comment here by `\State $span[I] \gets i$ \Comment{One Based Indexing}`